bin/yugabyted

```diff
@@ -526,6 +543,23 @@ class ControlScript(object):
                     " Removing...").format(data_dir_files, data_dir))
                 rmcontents(data_dir, exclude_names=[pid_file_name])
 
+            # Add enforced authentication flags on first installation
+            # Add ysql_enable_auth, if YSQL_PASSWORD env variable exists
+            # Add use_cassandra_authentication, if YCQL_USER or YCQL_PASSWORD env variables exists
+            if is_first_install:
+                # Add ysql_enable_auth flag to enforce authentication for YSQL
+                if self.setup_env_init.is_exists('YSQL_PASSWORD'):
+                    yb_tserver_cmd.extend(["--ysql_enable_auth=true"])
+                    self.configs.saved_data["ysql_auth"] = True
```

Comment:
can we instead add these to the cmd line flag tserver_flags, (which should automatically be persisted?)

Reply:
Yeah, I thought the same, although there is a scenario in which if the user spins up the YB server from the stopped state with some manipulation in the `tserver_flags` option then, this event override the flags in `yugabyted.conf` and spin up the server with the latest flags.
With the current implementation, they can't change the authentication flags with `yugabyted` options. If in case they want to change then, they can change manually in `yugabyted.conf`. This implementation reduces the chances of unintentional removal of authentication flags.

If still, we want to persist the flags in `tserver_flags` then we can remove these code snippets in the current implementation.
1. Lines: 477-483
```py
        # Add ysql_enable_auth flag to enforce authentication for YSQL
        if self.configs.saved_data.get("ysql_auth"):
            yb_tserver_cmd.extend(["--ysql_enable_auth=true"])

        # Add use_cassandra_authentication flag to enforce authentication for YCQL
        if self.configs.saved_data.get("ycql_auth"):
            yb_tserver_cmd.extend(["--use_cassandra_authentication=true"])
```
2. Lines: 1396-1397
```py
            "ysql_auth": DEFAULT_YSQL_AUTH,
            "ycql_auth": DEFAULT_YCQL_AUTH,
```
3. Lines: 97-98
```py
DEFAULT_YSQL_AUTH = False
DEFAULT_YCQL_AUTH = False
```